src/borg/constants.py

```diff
@@ -1,21 +1,76 @@
 # this set must be kept complete, otherwise the RobustUnpacker might malfunction:
-ITEM_KEYS = frozenset(['path', 'source', 'rdev', 'chunks', 'chunks_healthy', 'hardlink_master',
-                       'mode', 'user', 'group', 'uid', 'gid', 'mtime', 'atime', 'ctime', 'birthtime', 'size',
-                       'xattrs', 'bsdflags', 'acl_nfs4', 'acl_access', 'acl_default', 'acl_extended',
-                       'part'])
+ITEM_KEYS = frozenset(
+    [
+        "path",
+        "source",
+        "rdev",
+        "chunks",
+        "chunks_healthy",
+        "hardlink_master",
+        "mode",
+        "user",
+        "group",
+        "uid",
+        "gid",
+        "mtime",
+        "atime",
+        "ctime",
+        "birthtime",
+        "size",
+        "xattrs",
+        "bsdflags",
+        "acl_nfs4",
+        "acl_access",
+        "acl_default",
+        "acl_extended",
+        "part",
+    ]
+)
 
 # this is the set of keys that are always present in items:
-REQUIRED_ITEM_KEYS = frozenset(['path', 'mtime', ])
+REQUIRED_ITEM_KEYS = frozenset(
+    [
+        "path",
+        "mtime",
```

Comment:
There was a trailing comma after mtime. This would have stayed on one line:
```py3
REQUIRED_ITEM_KEYS = frozenset(['path', 'mtime'])
```

Reply:
I like trailing commas, they make future changes more "regular".

Reply:
It appears that trailing commas force black to split items across multiple lines:
```py3
REQUIRED_ITEM_KEYS = frozenset(['path', 'mtime',])
REQUIRED_ITEM_KEYS = frozenset(['path', 'mtime'])
```
becomes:
```py3
REQUIRED_ITEM_KEYS = frozenset(
    [
        "path",
        "mtime",
    ]
)
REQUIRED_ITEM_KEYS = frozenset(["path", "mtime"])
```

Reply:
Bug?

Reply:
Nope. This is a feature, but one that **can** be disabled. See https://github.com/psf/black/pull/1824

```sh
pipx run black .
```
produces:
```diff
diff --git a/example.py b/example.py
index 6e257e0..ed2a7da 100644
--- a/example.py
+++ b/example.py
@@ -1,2 +1,7 @@
-REQUIRED_ITEM_KEYS = frozenset(["path", "mtime",])
+REQUIRED_ITEM_KEYS = frozenset(
+    [
+        "path",
+        "mtime",
+    ]
+)
 REQUIRED_ITEM_KEYS = frozenset(["path", "mtime"])
```
but:
```sh
pipx run black --skip-magic-trailing-comma .
```
produces:
```diff
diff --git a/example.py b/example.py
index 6e257e0..3a4ce71 100644
--- a/example.py
+++ b/example.py
@@ -1,2 +1,2 @@
-REQUIRED_ITEM_KEYS = frozenset(["path", "mtime",])
+REQUIRED_ITEM_KEYS = frozenset(["path", "mtime"])
 REQUIRED_ITEM_KEYS = frozenset(["path", "mtime"])
```

Reply:
and what if we want to just keep the trailing comma and not have it split into one line per element?

it looks like the idea of "always do the same change" when adding or removing an element is not present in black:
```
[]  # start with an empty list
[first, ]  # add first element
[first, second, ]  # add another element - same procedure as first element.
```

Reply:
btw, for tuples, this also solves the non-tupleness of `(first)`, because then it is `(first, )`.

Reply:
I am sorry, but black does not have an option to keep a trailing comma when putting the output on a single-line. Is this a deal-breaker? We could theoretically ask for such an option to be added, but I am not sure it'll happen given that the default is [magic trailing comma](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma) - a trailing comma indicates that the output should be split across multiple lines.

Reply:
i actually like the fact that the comma is used as indicator on how one wants to be diff friendly/lined up

with the tailing comma black prepares multi-line, without the comma black prepares single line